tests/model_registry/model_catalog/utils.py

```
@@ -208,3 +208,124 @@ def _extract_properties_and_required(schema: dict[Any, Any]) -> tuple[set[str],
     }
 
     return all_properties - excluded_fields, required_fields - excluded_fields
+
+
+def validate_filter_options_structure(
+    response: dict[Any, Any], expected_properties: set[str] = None
```

Ruff (0.14.0)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)